### PR TITLE
Use ref_final_yr for model vs model

### LINKED ISSUE #666: [Bug]: Error from parameter checking: raise ParameterNotProvidedError(relevant_parameter)
### What happened?

When testing v3.0.0 rc2, I ran into error below. Not sure how this problem is introduced, but the checker is expecting  `ref_end_year` for `tropical_subseasonal` set  but what always provided as a parameter is `ref_final_year`.
```
Traceback (most recent call last):
  File "/usr/WS1/e3sm/apps/e3sm-unified/base/envs/e3sm_unified_1.11.0rc3_login/bin/zppy", line 8, in <module>
    sys.exit(main())
  File "/usr/WS1/e3sm/apps/e3sm-unified/base/envs/e3sm_unified_1.11.0rc3_login/lib/python3.10/site-packages/zppy/__main__.py", line 61, in main
    _launch_scripts(config, script_dir, job_ids_file, plugins)
  File "/usr/WS1/e3sm/apps/e3sm-unified/base/envs/e3sm_unified_1.11.0rc3_login/lib/python3.10/site-packages/zppy/__main__.py", line 237, in _launch_scripts
    existing_bundles = e3sm_diags(config, script_dir, existing_bundles, job_ids_file)
  File "/usr/WS1/e3sm/apps/e3sm-unified/base/envs/e3sm_unified_1.11.0rc3_login/lib/python3.10/site-packages/zppy/e3sm_diags.py", line 39, in e3sm_diags
    check_parameters_for_bash(c)
  File "/usr/WS1/e3sm/apps/e3sm-unified/base/envs/e3sm_unified_1.11.0rc3_login/lib/python3.10/site-packages/zppy/e3sm_diags.py", line 106, in check_parameters_for_bash
    check_required_parameters(c, set(["tropical_subseasonal"]), "ref_end_yr")
  File "/usr/WS1/e3sm/apps/e3sm-unified/base/envs/e3sm_unified_1.11.0rc3_login/lib/python3.10/site-packages/zppy/utils.py", line 241, in check_required_parameters
    raise ParameterNotProvidedError(relevant_parameter)
```

### What machine were you running on?

Ruby at LC. 

### Environment

zppy v3.0.0 rc2 within e3sm-unified v1.11.0rc3

### What command did you run?

```bash
zppy -c
```

### Copy your cfg file

```bash
[default]
#input = /p/lustre1/golaz1/E3SM/E3SMv3/20240828.v3.LR.piControl.ruby
output = /p/lustre1/zhang40/E3SMv3/post-processing/extendedOutput.v3.LR.historical_0101_rc3/
input = /usr/workspace/e3sm/zhang40/simulations/extendedOutput.v3.LR.historical_0101
#output = /usr/workspace/e3sm/zhang40/postprocessing/extendedOutput.v3.LR.historical_0101_try2
case = extendedOutput.v3.LR.historical_0101
www = /usr/global/web-pages/lc/www/e3sm/diagnostics_output/zhang40/E3SMv3/0130_rc3
mapping_file = map_ne30pg2_to_cmip6_180x360_traave.20231201.nc
machine = "ruby"
account = "e3smtest"
#partition = compute
partition = pbatch
environment_commands = "source /usr/workspace/e3sm/apps/e3sm-unified/load_latest_e3sm_unified_1.11.0rc3_ruby.sh"
#environment_commands = "source /g/g92/zhang40/miniforge3/etc/profile.d/conda.sh; conda activate e3sm-unified110"
#environment_commands = "source /lcrc/soft/climate/e3sm-unified/test_e3sm_unified_1.10.0rc5_chrysalis.sh"
campaign = "water_cycle"

[climo]
active = True
#years = "0001:0100:50", "0001:0100:100"
#years = "1985:2014:30", "1985:2014:15"
years = "2000:2014:15",
#years = "0001:0010:10",
walltime = "1:00:00"
#`bck` is needed because running NCO with `mpi` has error
parallel = "bck"

  [[ atm_monthly_180x360_aave ]]
  input_subdir = "archive/atm/hist"
  mapping_file = map_ne30pg2_to_cmip6_180x360_traave.20231201.nc
  frequency = "monthly"

  [[ atm_monthly_diurnal_8xdaily_180x360_aave ]]
  input_subdir = "archive/atm/hist"
  input_files = "eam.h3"
  mapping_file = map_ne30pg2_to_cmip6_180x360_traave.20231201.nc
  vars = "PRECT"
  frequency = "diurnal_8xdaily"

[ts]
active = True
years = "2000:2014:15"
walltime = "00:50:00"

  [[ atm_monthly_180x360_aave ]]
  input_subdir = "archive/atm/hist"
  input_files = "eam.h0"
  frequency = "monthly"
  mapping_file = /g/g92/zhang40/zppy_test/map_ne30pg2_to_cmip6_180x360_traave.20231201.nc
  vars = "FSNTOA,FLUT,FSNT,FLNT,FSNS,FLNS,SHFLX,QFLX,TAUX,TAUY,PRECC,PRECL,PRECSC,PRECSL,TS,TREFHT,CLDTOT,CLDHGH,CLDMED,CLDLOW,U,ICEFRAC,LANDFRAC,OCNFRAC,PS,CLDICE,CLDLIQ,T,AODDUST"
# Needed for mixed-phase partition
#  vars = "LANDFRAC,CLDICE,CLDLIQ,T"
  ts_fmt = "cmip"

  [[ atm_daily_180x360_aave ]]
  input_subdir = "archive/atm/hist"
  input_files = "eam.h1"
  frequency = "daily"
  mapping_file = /g/g92/zhang40/zppy_test/map_ne30pg2_to_cmip6_180x360_traave.20231201.nc
  # Needed for Wheeler Kiladis
  vars = "FLUT,PRECT,U850"

  [[ atm_monthly_glb ]]
  input_subdir = "archive/atm/hist"
  input_files = "eam.h0"
  frequency = "monthly"
  mapping_file = "glb"
[e3sm_diags]
active = True
walltime = "4:00:00"
#years = "0001:0100:50", "0001:0100:100"
years = "2000:2014:15",
ts_num_years = 15
ref_start_yr = 2000
ref_final_yr = 2014
multiprocessing = True
num_workers = 8

  [[ atm_monthly_180x360_aave ]]
#  environment_commands = "source /home/ac.zhang40/y/etc/profile.d/conda.sh; conda activate edv2110"
  short_name = 'v3.LR.historical_0101'
  grid = '180x360_traave'
  reference_data_path = '/usr/workspace/e3sm/data/diagnostics/observations/Atm/climatology'
  obs_ts = '/usr/workspace/e3sm/data/diagnostics/observations/Atm/time-series'
  dc_obs_climo = '/usr/workspace/e3sm/data/diagnostics/observations/Atm/climatology'
  climo_diurnal_subsection = "atm_monthly_diurnal_8xdaily_180x360_aave"
  climo_diurnal_frequency = "diurnal_8xdaily"
  ts_daily_subsection = "atm_daily_180x360_aave"
  sets="lat_lon","zonal_mean_xy","zonal_mean_2d","polar","cosp_histogram","meridional_mean_2d","annual_cycle_zonal_mean","qbo","diurnal_cycle","zonal_mean_2d_stratosphere","aerosol_aeronet","tropical_subseasonal",
```

### What jobs are failing?

```bash

```

### What stack trace are you encountering?

```bash

```

### PR DESCRIPTION
**Pull request by chengzhuzhang**
## Summary
The problem might be introduced from https://github.com/E3SM-Project/zppy/commit/abad6749dc398eac6dbfc445bb3a3af01b4e3fce. This PR is to revert the change regarding to `ref_end_yr`:
1. revert to use `ref_final_yr` as other variability sets.
2. remove checking this variable

Issue resolution:
- Closes #666 

Select one: This pull request is...
- [ ] a bug fix: increment the patch version
- [ ] a small improvement: increment the minor version
- [ ] a new feature: increment the minor version
- [ ] an incompatible (non-backwards compatible) API change: increment the major version

Please fill out either the "Small Change" or "Big Change" section (the latter includes the numbered subsections), and delete the other.

## Small Change

- [ ] To merge, I will use "Squash and merge". That is, this change should be a single commit.
- [ ] Logic: I have visually inspected the entire pull request myself.
- [ ] Pre-commit checks: All the pre-commits checks have passed.

## Big Change

- [ ] To merge, I will use "Create a merge commit". That is, this change is large enough to require multiple units of work (i.e., it should be multiple commits).

### 1. Does this do what we want it to do?

Required:
- [ ] Product Management: I have confirmed with the stakeholders that the objectives above are correct and complete.
- [ ] Testing: I have added or modified at least one "min-case" configuration file to test this change. Every objective above is represented in at least one cfg.
- [ ] Testing: I have considered likely and/or severe edge cases and have included them in testing.

If applicable:
- [ ] Testing: this pull request introduces an important feature or bug fix that we _must_ test often. I have updated the weekly-test configuration files, not just a "min-case" one.
- [ ] Testing: this pull request adds at least one new possible parameter to the cfg. I have tested using this parameter with and without any other parameter that may interact with it.

### 2. Are the implementation details accurate & efficient?

Required:
- [ ] Logic: I have visually inspected the entire pull request myself.
- [ ] Logic: I have left GitHub comments highlighting important pieces of code logic. I have had these code blocks reviewed by at least one other team member.

If applicable:
- [ ] Dependencies: This pull request introduces a new dependency. I have discussed this requirement with at least one other team member. The dependency is noted in `zppy/conda`, not just an `import` statement.

### 3. Is this well documented?

Required:
- [ ] Documentation: by looking at the docs, a new user could easily understand the functionality introduced by this pull request.

### 4. Is this code clean?

Required:
- [ ] Readability: The code is as simple as possible and well-commented, such that a new team member could understand what's happening.
- [ ] Pre-commit checks: All the pre-commits checks have passed.

If applicable:
- [ ] Software architecture: I have discussed relevant trade-offs in design decisions with at least one other team member. It is unlikely that this pull request will increase tech debt.

### CODE REVIEW

File: zppy/defaults/default.ini [hunk 1]
```diff
@@ -171,7 +171,6 @@ output_format = string_list(default=list("png"))
 # See https://e3sm-project.github.io/e3sm_diags/_build/html/master/available-parameters.html
 output_format_subplot = string_list(default=list())
 # End year (i.e., the last year to use) for the reference data
-# Required for "tropical_subseasonal" runs
 ref_end_yr = string(default="")
```
Comment (forsyth2) [inline zppy/defaults/default.ini:174]: We can probably just remove this then.

File: zppy/e3sm_diags.py [hunk 2]
```diff
@@ -103,7 +103,6 @@ def e3sm_diags(config: ConfigObj, script_dir: str, existing_bundles, job_ids_fil
 
 def check_parameters_for_bash(c: Dict[str, Any]) -> None:
     # Check parameters that aren't used until e3sm_diags.bash is run
-    check_required_parameters(c, set(["tropical_subseasonal"]), "ref_end_yr")
     check_required_parameters(c, set(["qbo"]), "ref_final_yr")
```
Comment (forsyth2) [inline zppy/e3sm_diags.py:106]: In `zppy/templates/e3sm_diags.bash`, we have:
```bash
{% if run_type == "model_vs_obs" %}
# Obs
trop_param.reference_data_path = '{{ obs_ts }}'
trop_param.ref_start_yr = 2001
trop_param.ref_end_yr = 2010
```
Those are hard coded, so we're good to delete this check here, yes

File: zppy/templates/e3sm_diags.bash [hunk 3]
```diff
@@ -287,8 +287,8 @@ params.append(enso_param)
 trop_param = TropicalSubseasonalParameter()
 trop_param.test_data_path = '${ts_daily_dir}'
 trop_param.test_name = short_name
-trop_param.test_start_yr = f'{start_yr:04}'
-trop_param.test_end_yr = f'{end_yr:04}'
```
Comment (forsyth2) [inline zppy/templates/e3sm_diags.bash:291]: I think these are always put in as 4-digit years, so this change should be ok

**Review by forsyth2 — COMMENTED** [review]
@chengzhuzhang Ok, I think these changes look good. The unit tests and integration test cfgs will need to be updated though. I can add a commit for those changes if you'd like. Thanks.

**Comment by chengzhuzhang** [comment]
@forsyth2 thanks for a review. Could you go ahead to update the tests? Thanks.

**Comment by forsyth2** [comment]
Added a 3rd commit, will use merge commit to preserve authorship of the different commits. Unit tests pass.
